### PR TITLE
Update Variant Example Application

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
## Why this PR?

Variant feature flags can be used in conjunction with dependency injection to surface different implementations of a service for different users.

This PR adds an example to demonstrate how to use `VariantServiceProvider<T>` to inject different implementations of an interface according to the assigned variant.

The example application is a web app that allows users to enter two numbers and the web app will calculate the sum. There are two variants of the calculation service: one will do the calculation on the server locally and the other will call an external API to do the calculation. This web app will do A/B test for these two variants.

For more details, please refer to the included README.

## Visible Changes

- Add VariantServiceDemo project
- Remove unused code and reference for EvaluationDataToApplicationInsights example

### CODE REVIEW

File: examples/VariantServiceDemo/appsettings.json
Comment (jimmyca15) [inline examples/VariantServiceDemo/appsettings.json:37]:
```suggestion
              "from": 50,
```

File: examples/VariantServiceDemo/appsettings.json
Comment (jimmyca15) [inline examples/VariantServiceDemo/appsettings.json:10]: Remove

File: examples/VariantServiceDemo/HttpContextTargetingContextAccessor.cs
Comment (jimmyca15) [inline examples/VariantServiceDemo/HttpContextTargetingContextAccessor.cs:34]: How does the username cookie get set?

**Review by jimmyca15 — APPROVED** [review]

**Comment by zhiyuanliang-ms** [comment]
@jimmyca15 I will merge this PR after #461 is merged. In this case, I can also add the example reference for README in this PR.

File: examples/VariantServiceDemo/appsettings.json
Comment (zhiyuanliang-ms) [reply examples/VariantServiceDemo/appsettings.json:10]: Sorry for the mistake. That app insights store should have been deleted.

**Pull request merged**
